src/components/Api.js

```diff
@@ -0,0 +1,131 @@
+export default class Api {
+  constructor({ baseURL, authToken }) {
+    this._baseURL = baseURL;
+    this._authToken = authToken;
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseURL}//users/me`, {
+      headers: {
+        authorization: this._authToken,
+      },
+    })
+      .then((res) => {
+        return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.